### ISSUE TITLE
Misleading errors

### ISSUE THREAD

**Issue by FedericoCeratto** [Yubico/libu2f-server#29]
Not setting public key before calling u2fs_authentication_verify leaves ctx->key empty leading to a very cryptic memory error. 
There are other instances of generic use of U2FS_MEMORY_ERROR e.g. https://github.com/Yubico/libu2f-server/blob/master/u2f-server/core.c#L1191
It would be really helpful to print out more explicit error messages e.g. "The public key has not been set" to allow library users debug their code.

**Comment by thorduri** [comment]
You are correct, these errors are cryptic and we should introduce new error codes for invalid arguments, e.g. `U2FS_ARGUMENT_ERROR` and review the current debugging output.

This would necessarily need to be a part of a minor release (#27).

Thanks for the heads up!

**Comment by thorduri** [comment]
@FedericoCeratto For my records, are you maintaining an open source application using the library ?

If so I'd appreciate a link!
